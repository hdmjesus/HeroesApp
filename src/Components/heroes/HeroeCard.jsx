import React from "react";
import { Link } from "react-router-dom";

export const HeroeCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {
  return (
    <div
      className="card mt-5 mb-3 m-1 "
      style={{ maxWidth: 350, padding: 0, minWidth: 150 }}
    >
      <div className="row g-0">
        <div className="col-md-4">
          <img
            src={`./images/${id}.webp`}
            className="card-img"
            alt="superhero"
          />
        </div>
        <div className="col-md-8">
          <div className="card-body">
            <h5 className="card-title">{superhero}</h5>
            <p className="card-text">{alter_ego}</p>
            {alter_ego !== characters && <p className=""> {characters}</p>}
            <p className="card-text">
              {" "}
              <small className="text-muted"> {first_appearance}</small>
            </p>
            <Link to={`./hero/${id}`}>Mas...</Link>
          </div>
        </div>
      </div>
    </div>
  );
};
