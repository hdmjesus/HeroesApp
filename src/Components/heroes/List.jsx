import React from "react";
import { getHeroByPublisher } from "../../selectors/getHeroesByPublisher";
import { HeroeCard } from "./HeroeCard";

export const HeroesList = ({ publisher }) => {
  const heroes = getHeroByPublisher(publisher);
  return (
    <div className="flex justify-center flex-wrap animate__animated animate__fadeIn">
      {heroes.map((hero) => (
        <HeroeCard key={hero.id} {...hero} />
      ))}
    </div>
  );
};
