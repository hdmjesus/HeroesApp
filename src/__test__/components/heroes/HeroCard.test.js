import { mount } from "enzyme";
import React from "react";
import { HeroesList } from "../../../Components/heroes/List";
import { HeroeCard } from "../../../Components/heroes/HeroeCard";
import { MemoryRouter } from "react-router";
describe("Pruebas en el componente <HeroCard/>", () => {
  test("debe renderizarce correctamente el card", () => {
    const batman = {
      id: "dc-batman",
      superhero: "Batman",
      publisher: "DC Comics",
      alter_ego: "Bruce Wayne",
      first_appearance: "Detective Comics #27",
      characters: "Bruce Wayne",
    };
    const wrapper = mount(
      <MemoryRouter>
        <HeroeCard {...batman} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".card-title").text().trim()).toBe(batman.superhero);
  });
});
