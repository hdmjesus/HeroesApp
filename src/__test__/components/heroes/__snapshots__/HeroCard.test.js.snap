// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HeroCard/> debe renderizarce correctamente el card 1`] = `
<div
  className="card mt-5 mb-3 m-1  text-xs"
  style={
    Object {
      "maxWidth": 350,
      "minWidth": 150,
      "padding": 0,
    }
  }
>
  <div
    className="row g-0"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="superhero"
        className="card-img"
        src="./images/dc-batman.webp"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Batman
        </h5>
        <p
          className="card-text"
        >
          Bruce Wayne
        </p>
        <p
          className="card-text"
        >
           
          <small
            className="text-muted"
          >
             
            Detective Comics #27
          </small>
        </p>
        <a
          className="text-blue-500"
          href="/hero/dc-batman"
          onClick={[Function]}
        >
          Mas...
        </a>
      </div>
    </div>
  </div>
</div>
`;
